### DISCUSSION TITLE
sendZclFrameToEndpointInternal error with Sonoff ZBDongle-E

### DISCUSSION THREAD

**Discussion by Ferywir** [Koenkk/zigbee2mqtt#18154] [Devices]
Good morning,

I've had these bugs for a while now, it works overall, but from time to time I have a light that stays on or with a different intensity than the others.

here is the config:

Jeedom with JeeZigbee (Zigbee2mqtt)
Sonof ZBDongle-E
Version: 1.31.2
Coordinator: EZSP v8
Port: /dev/ttyACM1

I re-included all the disturbing elements, it worked until I updated the version.

do you have an idea ?

Thanks
Ferywir

`Zigbee2MQTT:error 2023-06-30 14:01:25: Publish 'set' 'color_temp' to 'Hue white ambiance - SdD - 04' failed: 'Error: Command 0x001788010c9fa72f/11 lightingColorCtrl.moveToColorTemp({"colortemp":229,"transtime":0}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (sendZclFrameToEndpointInternal error)'
Zigbee2MQTT:error 2023-06-30 14:09:06: Publish 'set' 'color_temp' to 'Hue white ambiance - SdD - 04' failed: 'Error: Command 0x001788010c9fa72f/11 lightingColorCtrl.moveToColorTemp({"colortemp":229,"transtime":0}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (sendZclFrameToEndpointInternal error)'
Zigbee2MQTT:error 2023-06-30 14:10:06: Publish 'set' 'color_temp' to 'Hue white ambiance - SdD - 02' failed: 'Error: Command 0x001788010bc3e0f3/11 lightingColorCtrl.moveToColorTemp({"colortemp":229,"transtime":0}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (sendZclFrameToEndpointInternal error)'
Zigbee2MQTT:error 2023-06-30 14:10:06: Publish 'set' 'color_temp' to 'Hue white ambiance - SdD - 06' failed: 'Error: Command 0x001788010bbdd6b9/11 lightingColorCtrl.moveToColorTemp({"colortemp":229,"transtime":0}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (sendZclFrameToEndpointInternal error)'
Zigbee2MQTT:error 2023-06-30 14:27:40: Publish 'set' 'brightness' to 'Hue white ambiance - Couloir nord - 02' failed: 'Error: Command 0x0017880108493259/11 genLevelCtrl.moveToLevelWithOnOff({"level":254,"transtime":0}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (sendZclFrameToEndpointInternal error)'
Zigbee2MQTT:error 2023-06-30 14:27:44: Publish 'set' 'color_temp' to 'Hue white ambiance - SdD - 06' failed: 'Error: Command 0x001788010bbdd6b9/11 lightingColorCtrl.moveToColorTemp({"colortemp":229,"transtime":0}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (sendZclFrameToEndpointInternal error)'
Zigbee2MQTT:error 2023-06-30 15:04:01: Publish 'set' 'state' to 'Hue white ambiance - Couloir sud - 02' failed: 'Error: Command 0x0017880108493112/11 genLevelCtrl.moveToLevelWithOnOff({"level":0,"transtime":0}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (sendZclFrameToEndpointInternal error)'`

**Comment by PetrMa** [comment]
I have the same errors last 3 days on CC2531 and latest version of HA:

Zigbee2MQTT:error 2023-07-03 14:38:07: Publish 'set' 'system_mode' to 'Topení Tomáš' failed: 'Error: Command 0x003c84fffec7f78d/1 manuSpecificTuya.dataRequest({"seq":5,"dpValues":[{"dp":106,"datatype":4,"data":[2]}]}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'No network route' (205))'
Zigbee2MQTT:info  2023-07-03 14:38:07: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'system_mode' to 'Topení Tomáš' failed: 'Error: Command 0x003c84fffec7f78d/1 manuSpecificTuya.dataRequest({\"seq\":5,\"dpValues\":[{\"dp\":106,\"datatype\":4,\"data\":[2]}]}, {\"sendWhen\":\"immediate\",\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":true,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (Data request failed with error: 'No network route' (205))'","meta":{"friendly_name":"Topení Tomáš"},"type":"zigbee_publish_error"}'

**Comment by Ferywir** [comment]
ok so it certainly does not come from the controller.

**Comment by Ferywir** [comment]
Does anyone have an idea or have the same problem?

I tried several things:
- Update Zigbee2MQTT (1.32)
- Disconnect the lamps and reconnect them (power failure)
- Disconnect and reconnect the zigbee key
- restart Zigbee2MQTT
- restart the domotic server (jeedom)
- Randomly repair a few bulbs
- Turn off Wifi (2.4ghz and 5ghz)

Which is weird, I have a total of 5 groups of bulbs:
1) 3 bulbs
2) 4 bulbs
3) 2 bulbs
4) 6 bulbs
5) 6 bulbs
6) 1 bulbs

I have no errors in area 1/2/5.

here is a layout map of my lamps:
![Groupe lumières](https://github.com/Koenkk/zigbee2mqtt/assets/65760459/aabb8779-74f3-4eff-9699-8cd540bc10f3)


my zigbee controller (Sonoff ZBDongle-E) is in room/group1

thanks in advance everyone.
Ferywir

**Comment by Ferywir** [comment]
after 2 hours of testing ... (various and varied) :-)

I noticed something very interesting.
when I send an ignition command to my home automation system I send 3 commands at the same time:
- Ignition
- Color temperature
- Brightness

I added 2 delays between my 3 commands (0.3sec => 300ms) and magic no more errors!

However, I find it odd, it worked for weeks ... without any errors, and following an update (of memory from 1.30.4 to 1.31.2), I started to have these errors ...

Do I have a router (lamp) that is corrupted? how to find it?

THANKS
Good day

**Comment by lanvideosat** [comment]
У меня такая же ошибка, подскажите дело в самом ключ zigbee или нет?

**Comment by Ferywir** [comment]
I have no idea, I'm just trying to figure out where the problem is.

according to my tests it is I think the sending of several commands at the same time to the equipment (power on, color, brightness). I suppose it bugs when it tries for example to change the color before having received the ON command. which is bugging.

by adding 300ms delays between my commands I solve the errors. but I don't know why it worked before and mtn no longer? corrupted equipment? dongle? ....

I would like to try with another dongle (the same sonoff ZBDongle-E) but I don't know how to do it, I didn't get my answer (https://github.com/Koenkk/zigbee2mqtt/discussions/18079) .

THANKS

**Comment by ZigWave5743** [comment]
On the TuYa PJ-MGW1203.
Zigbee2MQTT    [1.31.2]
Coordinator type    EZSP v8
Coordinator revision    6.10.3.0 build 297
Coordinator IEEE Address    0xe0798dfffed20e24
Frontend version    0.6.129

The device will run for a couple of days, then spew out 600+ messages.  Have to stop/start ZigBee2MQTT to restore the device

Z2M Server: Exception while calling fromZigbee converter: Command 0xf082c0fffe18cf92/1 manuSpecificTuya.mcuGatewayConnectionStatus({"payloadSize":1,"payload":1}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (sendZclFrameToEndpointInternal error)}

**Comment by nicolinuxfr** [comment]
Same issue for me, using the Dongle-E from Sonoff. I'm noticing the error much frequently when I crossed 40 devices, so I'm wondering if it's not an issue related to the size of the network.

Anyway, I'm thinking about using another, more powerful, coordinator. Unfortunately, it will mean I have to start over, which is not something I look forward, as you can guess.

I tried to create an automation based on the Z2M logs (and then restart the module using a service), but could not find a way. If someone has an idea, it would be better in the meantime.

**Reply by thaynes43** [reply]
I am struggling with a pi4 and a Sonoff ZBDongle-E dedicated to Z2M. I have 71 devices right now and ever couple of days nothing responds and Z2M needs to be restarted to get things going again. I can't find any point in the logs where it looks like it froze up, I just start seeing errors like these when I try to toggle things:

```
error 2023-12-21 19:51:21: Publish 'set' 'brightness' to 'Entrance Bottom Hue Bulb 2' failed: 'Error: Command 0x001788010db1e967/11 genLevelCtrl.moveToLevelWithOnOff({"level":254,"transtime":0}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Timeout - 22653 - 11 - 234 - 8 - 11 after 10000ms)'

error 2023-12-21 19:59:30: Publish 'set' 'state' to 'Upstairs Hallway Bulb 1' failed: 'Error: Command 0x0017880103b1891e/11 [genOnOff.off](https://genonoff.off/)({}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (sendZclFrameToEndpointInternal error)'

error 2023-12-21 20:00:16: Publish 'set' 'brightness' to 'Upstairs Hallway Bulb 1' failed: 'Error: Command 0x0017880103b1891e/11 genLevelCtrl.moveToLevelWithOnOff({"level":202,"transtime":0}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (sendZclFrameToEndpointInternal error)'
```

Not sure how I could detect this automatically without getting really janky and triggering a switch on a polling loop, then parsing the log for the error and if I see it restart Z2M... Rather figure out why it's not reliable and get that sorted out.

**Reply by nicolinuxfr** [reply]
I tried to be clever and automate a Z2M restart each night at 3am. Unfortunately, I just noticed it means I miss some data, as if the network was down for several hours. 

<img width="683" alt="Capture d’écran 2024-01-03 à 09 15 00" src="https://github.com/Koenkk/zigbee2mqtt/assets/2078386/238c1312-b03e-4e50-ba2c-0dc05873cc53">

And even with this automation, I still have to restart Z2M each day, often multiple times per day. I think I want to get rid of this dongle, even if transitioning away will be painful but right now, all my house feels constantly broken because of this issue.

**Comment by nicolinuxfr** [comment]
Is it just for me or is this bug way worse with latest versions ? My ZigBee network basically stops working multiple times per hour, and even restarting Z2M is not enough. I don't know what to do, except start over with a new coordinator.

<img width="1664" alt="Capture d’écran 2024-01-04 à 13 23 47" src="https://github.com/Koenkk/zigbee2mqtt/assets/2078386/aab28435-d521-4ee7-ad16-be9fb1d1f8cd">

**Comment by Shellazo** [comment]
Same here...it's been ongoing for a few weeks, recently has become much more frequent. I'm using Skyconnect.

<img width="1938" alt="Screenshot 2024-01-04 at 9 14 15 AM" src="https://github.com/Koenkk/zigbee2mqtt/assets/36079465/37834587-daca-4387-b6dd-eef5510ecdb7">

**Comment by felipejgribeiro** [comment]
For me is the same. Recently has become much more frequent since I updated z2m in Home Assistant to 1.35.1

**Reply by NicoBenndorf** [reply]
Same thing and error "sendZclFrameToEndpointInternal error" in the last few days I have to restart z2m multiple times per day..

**Comment by ashulgach** [comment]
Same happening for me. Downgraded to 1.34.0 and stable overnight, at least, but we'll see how long it lasts. Running in a docker container instead of an add-on made it easier to do the downgrade.

**Comment by nberardi** [comment]
I am having the exact same issue, I am on 1.35.1-1 with Skyconnect running in multiprotocol.  I can restart the system and it starts running normally again, but after a few hours I start receiving these errors again.  I have also tried Edge as well, with the same exact result.

**Comment by emiham** [comment]
I had this issue too with the Dongle-E. It seems to have been solved by upgrading the firmware to the latest version.

**Reply by XFNeo** [reply]
What firmware version did you use and what version did you update to?
